File: Assets/Game/Script/Net/Api/UserAgent.cs
using System;
using UnityEngine;

namespace ZJYFrameWork.Net
{
    public class UserAgent
    {
        private static UserAgent instance;

        public static string Value
        {
            get
            {
                instance = instance ?? new UserAgent();
                return instance.ToString();
            }
        }

        private string cached;

        private UserAgent()
        {
            string appName = Application.identifier;
            string version = Application.version;
            string operatingSystem = SystemInfo.operatingSystem;
            string device = SystemInfo.deviceModel;

            cached = $"{appName}/{version} ({operatingSystem}; {device})";

#if UNITY_EDITOR
            var platform = "Unknown";
            var buildTarget = UnityEditor.EditorUserBuildSettings.activeBuildTarget;
            platform = buildTarget switch
            {
                UnityEditor.BuildTarget.iOS => "iPhone",
                UnityEditor.BuildTarget.Android => "Android",
                _ => platform
            };
            cached += $"<Emulating:{platform}>";
#endif
        }

        public override string ToString()
        {
            return cached;
        }
    }
}

File: Lua/appliction/app/net/http/userAgent.lua
UserAgent = class("UserAgent")
local cached = string.empty

local appVersion = string.empty

function UserAgent:Value()
    if string.IsNullOrEmtty(cached) then
        local appName = CS.UnityEngine.Application.identifier
        local version = CS.UnityEngine.Application.version
        local operatingSystem = CS.UnityEngine.SystemInfo.operatingSystem
        local device = CS.UnityEngine.SystemInfo.deviceModel
        cached = appName .. "/" .. version .. "(" .. operatingSystem .. "; " .. device .. ")"
    end
    return cached
end

---@return string
function UserAgent:Version()
    if string.IsNullOrEmtty(appVersion) then
        appVersion = CS.UnityEngine.Application.version
    end
    return appVersion
end
